drivers/ethernet/eth_stm32_hal.c

```diff
@@ -812,6 +812,10 @@ void HAL_ETH_TxCpltCallback(ETH_HandleTypeDef *heth_handle)
 	k_sem_give(&dev_data->tx_int_sem);
 
 }
+#endif /* CONFIG_SOC_SERIES_STM32H7X || CONFIG_ETH_STM32_HAL_API_V2 */
+
+#if defined(CONFIG_ETH_STM32_HAL_API_V2)
+#elif defined(CONFIG_SOC_SERIES_STM32H7X)
```

Comment:
Let's move this to the next commit for simplicity.

Reply:
Okay